conans/test/utils/tools.py

```diff
@@ -659,14 +661,36 @@ def scm_info_cache(self, reference):
 
         if not isinstance(reference, ConanFileReference):
             reference = ConanFileReference.loads(reference)
-        layout = self.cache.package_layout(reference)
+        layout = self.get_latest_ref_layout(reference)
         content = load(layout.conandata())
         data = yaml.safe_load(content)
         if ".conan" in data:
             return self._create_scm_info(data[".conan"])
         else:
             return self._create_scm_info(dict())
 
+    def get_latest_prev(self, ref: ConanReference, package_id=None) -> PackageReference:
+        """Get the latest PackageReference given a ConanReference"""
+        latest_rrev = self.cache.get_latest_rrev(ref)
+        if package_id:
+            pref = PackageReference(latest_rrev, package_id)
+        else:
+            pref = self.cache.get_package_ids(latest_rrev)[0]
```

Comment:
This is a bit dangerous. I understand the utility, but please, at least raise an error if there is more than 1 package_id in the package.